daemon/repo-mgr.c

```diff
@@ -3708,6 +3710,8 @@ apply_worktree_changes_to_index (SeafRepo *repo, struct index_state *istate,
              */
             if (!is_path_writable(repo->id,
                                   repo->is_readonly, event->path)) {
+                send_file_sync_error_notification (repo->id, repo->name, event->path,
+                                                   SYNC_ERROR_ID_SHARE_REPO_READ_ONLY);
```

Comment:
There is another place needs to send this notification. That's in the add_file() function. It checks whether the path of a file is writable before adding it to index.

When the daemon restarts, it scan the whole worktree recursively. In that case, the permission check is only done in add_file() function.